drivers/src/key_vault.rs

```diff
@@ -141,6 +141,42 @@ bitfield! {
     pub ecc_data, set_ecc_data:5;
 }
 
+pub trait KeyUsageBuilder {
+    fn set_hmac_key_en(&mut self) -> KeyUsage;
+    fn set_hmac_data_en(&mut self) -> KeyUsage;
+    fn set_sha_data_en(&mut self) -> KeyUsage;
+    fn set_ecc_private_key_en(&mut self) -> KeyUsage;
+    fn set_ecc_key_gen_seed_en(&mut self) -> KeyUsage;
+    fn set_ecc_data_en(&mut self) -> KeyUsage;
+}
+
+impl KeyUsageBuilder for KeyUsage {
```

Comment:
Why not `impl KeyUsage {`, and get rid of the trait?

Reply:
Thanks, done.